### PR TITLE
[Chrome] Add chrome.storage.getKeys() API

### PR DESCRIPTION
**Pull request by tomvangoethem**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Chrome Developers reference](https://developer.chrome.com/docs/extensions/reference/api/storage#method-StorageArea-getKeys)

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/chrome/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5a38000ae8461a409e339191e0e7dff1df7437bd
</p>

**Comment by tomvangoethem** [comment]
Friendly ping to review this PR.

File: types/chrome/test/index.ts [hunk 1]
```diff
@@ -848,6 +848,13 @@ function testStorage() {
         var myNewValue: { x: number } = changes["myKey"].newValue;
         var myOldValue: { x: number } = changes["myKey"].oldValue;
     });
+
+    function getKeysCallback(keys: string[]) {
+        console.log(keys);
+    }
+
+    chrome.storage.sync.getKeys();
+    chrome.storage.sync.getKeys(getKeysCallback);
```
Comment (erwanjugand) [inline types/chrome/test/index.ts:857]:
```suggestion
    chrome.storage.sync.getKeys(); // $ExpectType Promise<string[]>
    chrome.storage.sync.getKeys((keys) => { // $ExpectType void
      keys;  // $ExpectType string[]
    });
```

**Review by erwanjugand — COMMENTED** [review]
A little suggestion 😊

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/chrome/test/index.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e634a8280789d613afe828984eed0f0d9ed714d9
</p>

**Review by erwanjugand — APPROVED** [review]

File: types/chrome/test/index.ts [hunk 2]
```diff
@@ -848,6 +848,13 @@ function testStorage() {
         var myNewValue: { x: number } = changes["myKey"].newValue;
         var myOldValue: { x: number } = changes["myKey"].oldValue;
     });
+
+    chrome.storage.sync.getKeys(); // $ExpectType Promise<string[]>
+    chrome.storage.sync.getKeys((keys) => { // $ExpectType void
+      keys;  // $ExpectType string[]
```
Comment (erwanjugand) [inline types/chrome/test/index.ts:854]: There is just a problem with the indentation of my suggestion 😅
```suggestion
        keys;  // $ExpectType string[]
```

**Review by erwanjugand — APPROVED** [review]

**Review by iisaduan — APPROVED** [review]

**Comment by tomvangoethem** [comment]
Ready to merge

**Pull request merged**